packager/VisibilityChecker.cc

```diff
@@ -496,31 +512,140 @@ class VisibilityCheckerPass final {
                     if (pkgName.exists()) {
                         core::Context ctx{gs, core::Symbols::root(), f.file};
                         VisibilityCheckerPass pass{ctx, gs.packageDB().getPackageInfo(pkgName)};
                         ast::TreeWalk::apply(ctx, pass, f.tree);
+                        for (auto &[package, packagesToImport] : pass.toImport) {
+                            for (auto &[packageToImport, isTestImport] : packagesToImport) {
+                                importCapacitor.emplace_back(package, packageToImport, isTestImport);
+                            }
+                        }
+
+                        for (auto &[package, packagesToReplace] : pass.convertTestImport) {
+                            for (auto pair : packagesToReplace) {
+                                replaceTestImportCapacitor.emplace_back(package, pair.first, pair.second);
+                            }
+                        }
                     }
                 }
             }
 
-            barrier.DecrementCount();
+            if (processedByThread > 0) {
+                toImportQ->push(std::move(importCapacitor), processedByThread);
+                replaceTestImportQ->push(std::move(replaceTestImportCapacitor), processedByThread);
+            }
         });
 
-        barrier.Wait();
+        UnorderedMap<core::packages::MangledName, PackageFixes> combinedFixes;
+        std::vector<std::tuple<core::packages::MangledName, core::packages::MangledName, bool>> importThreadResult;
+        for (auto result = toImportQ->wait_pop_timed(importThreadResult, WorkerPool::BLOCK_INTERVAL(), gs.tracer());
+             !result.done();
+             result = toImportQ->wait_pop_timed(importThreadResult, WorkerPool::BLOCK_INTERVAL(), gs.tracer())) {
+            if (!result.gotItem()) {
+                continue;
+            }
+            for (auto &[package, packageToImport, isTestImport] : importThreadResult) {
+                combinedFixes[package].toImport.emplace(packageToImport, isTestImport);
+            }
+        }
 
+        std::vector<std::tuple<core::packages::MangledName, core::packages::MangledName, core::Loc>>
+            replaceTestImportThreadResult;
+        for (auto result = replaceTestImportQ->wait_pop_timed(replaceTestImportThreadResult,
+                                                              WorkerPool::BLOCK_INTERVAL(), gs.tracer());
+             !result.done(); result = replaceTestImportQ->wait_pop_timed(replaceTestImportThreadResult,
+                                                                         WorkerPool::BLOCK_INTERVAL(), gs.tracer())) {
+            if (!result.gotItem()) {
+                continue;
+            }
+            for (auto &[package, packageToImport, loc] : replaceTestImportThreadResult) {
+                combinedFixes[package].convertTestImports.emplace(packageToImport, loc);
+            }
+        }
```

Comment:
Is there a way we can do this with one queue? I'm worried about having one queue grow very large without anything consuming from it until the first queue completely drains.